src/components/Hamburger/Button/HamburgerButton.js

```diff
@@ -1,14 +1,48 @@
 import React, { Component } from 'react';
 
+import { ReactReduxContext, connect } from 'react-redux';
+
 import { media, colors, header } from '../../theme';
+import { toggleHamburgerMenu as toggleHamburgerMenuAction } from '../../../state/app';
 
 class HamburgerButton extends Component {
+
+  static contextType = ReactReduxContext;
+
   constructor(props) {
     super(props);
+
+    this.mediaQueryListener = null;
+    this.onMatchMQ = mediaQueryListener => this._onMatchMQ(mediaQueryListener);
+    this.toggleBodyStyle = toggle => this._toggleBodyStyle(toggle);
+  }
+
+  componentDidMount = () => {
+    if (!window.matchMedia) return;
```

Comment:
Even IE10 supports window.matchMedia, why we need this?

Reply:
Done